internal/secret-storage-backend/server.go

```diff
@@ -268,6 +269,40 @@ func (h *Handler) OnDelete(_ context.Context, request *pb.OnDeleteRequest) (*pb.
 	return &pb.OnDeleteResponse{}, nil
 }
 
+// OnDeleteRevision handles TypeInstance's revision deletion by removing a secret entry in a given provider.
+// It checks whether a given TypeInstance is locked before doing such operation.
+func (h *Handler) OnDeleteRevision(ctx context.Context, request *pb.OnDeleteRevisionRequest) (*pb.OnDeleteRevisionResponse, error) {
+	if request == nil {
+		return nil, NilRequestInputError
+	}
+
+	provider, err := h.getProviderFromContext(request.Context)
+	if err != nil {
+		return nil, err
```

Comment:
should we use `errors.Wrap`?

Reply:
nope, in other places we also don't use it as the `getProviderFromContext` already returns proper grpc errors, and we don't want to override them